server/char/list.py

```diff
@@ -102,38 +143,44 @@ def __init__(self, character_id, unknown1, character_name, character_unapproved_
 
 
     def serialize(self, stream):
-        stream.write(c_int64(character_id))
-        stream.write(c_uint32(unknown1))
-        stream.write(character_name)
-        stream.write(character_unapproved_name)
-        stream.write(is_name_rejected)
-        stream.write(free_to_play)
-        stream.write(unknown2, allocated_length=10)
-        stream.write(c_uint32(shirt_color))
-        stream.write(c_uint32(shirt_style))
-        stream.write(c_uint32(pants_color))
-        stream.write(c_uint32(hair_style))
-        stream.write(c_uint32(hair_color))
-        stream.write(c_uint32(lh))
-        stream.write(c_uint32(rh))
-        stream.write(c_uint32(eyebrows))
-        stream.write(c_uint32(eyes))
-        stream.write(c_uint32(mouth))
-        stream.write(c_uint32(unknown3))
-        stream.write(c_uint16(last_zone))
-        stream.write(c_uint16(last_instance))
-        stream.write(c_uint32(last_clone))
-        stream.write(c_uint64(last_login))
-        stream.write(c_uint16(len(items)))
-        for item in items:
+        """
+        Serializes the packet
+        """
+        stream.write(c_int64(self.character_id))
+        stream.write(c_uint32(self.unknown1))
+        stream.write(self.character_name)
+        stream.write(self.character_unapproved_name)
+        stream.write(self.is_name_rejected)
+        stream.write(self.free_to_play)
+        stream.write(self.unknown2, allocated_length=10)
+        stream.write(c_uint32(self.shirt_color))
+        stream.write(c_uint32(self.shirt_style))
+        stream.write(c_uint32(self.pants_color))
+        stream.write(c_uint32(self.hair_style))
+        stream.write(c_uint32(self.hair_color))
+        stream.write(c_uint32(self.lh))
+        stream.write(c_uint32(self.rh))
+        stream.write(c_uint32(self.eyebrows))
+        stream.write(c_uint32(self.eyes))
+        stream.write(c_uint32(self.mouth))
+        stream.write(c_uint32(self.unknown3))
+        stream.write(c_uint16(self.last_zone))
+        stream.write(c_uint16(self.last_instance))
+        stream.write(c_uint32(self.last_clone))
+        stream.write(c_uint64(self.last_login))
+        stream.write(c_uint16(len(self.items)))
+        for item in self.items:
             stream.write(c_uint32(item))
 
     @classmethod
     def deserialize(cls, stream):
+        """
+        Deserializes the packet
+        """
         character_id = stream.read(c_int64)
         unknown1 = stream.read(c_uint32)
-        character_name = stream.read(CString(self.unknown, allocated_length=66))
-        character_unapproved_name = stream.read(CString(self.unknown, allocated_length=66))
+        character_name = stream.read(CString('', allocated_length=66))
```

Comment:
This is definitely not the way it should go (fault of mine)... Thoughts about adding a classmethod to CString like `CString.withLength`, which doesn't require data, and just initializes it to `''` or something?

Reply:
Could do that yeah